src/healthcheck.c

```diff
@@ -457,52 +457,77 @@ bool healthcheck_parse_oci_annotations(const char *annotations_json, healthcheck
         return false;
     }
     
-    /* Set defaults */
+    /* Initialize config - no defaults, all values must be provided by Podman */
     config->enabled = true;
-    config->interval = 30;      // Default 30 seconds
-    config->timeout = 10;       // Default 10 seconds
-    config->start_period = 0;   // Default 0 seconds
-    config->retries = 3;        // Default 3 retries
+    config->interval = 0;
+    config->timeout = 0;
+    config->start_period = 0;
+    config->retries = 0;
+    config->test = NULL;
     
-    /* Parse Test command */
+    /* Parse Test command - REQUIRED */
     cJSON *test_array = cJSON_GetObjectItem(json, "test");
-    if (cJSON_IsArray(test_array) && cJSON_GetArraySize(test_array) >= 2) {
-        cJSON *cmd_type = cJSON_GetArrayItem(test_array, 0);
-        cJSON *cmd_value = cJSON_GetArrayItem(test_array, 1);
-        
-        if (cJSON_IsString(cmd_type) && cJSON_IsString(cmd_value)) {
-            if (strcmp(cmd_type->valuestring, "CMD") == 0 || strcmp(cmd_type->valuestring, "CMD-SHELL") == 0) {
-                /* Create test command array */
-                config->test = calloc(2, sizeof(char*));
-                config->test[0] = strdup(cmd_value->valuestring);
-                config->test[1] = NULL;
-            }
-        }
+    if (!cJSON_IsArray(test_array) || cJSON_GetArraySize(test_array) < 2) {
+        nwarn("Healthcheck configuration missing required 'test' command");
+        cJSON_Delete(json);
+        return false;
+    }
+    
+    cJSON *cmd_type = cJSON_GetArrayItem(test_array, 0);
+    cJSON *cmd_value = cJSON_GetArrayItem(test_array, 1);
+    
+    if (!cJSON_IsString(cmd_type) || !cJSON_IsString(cmd_value)) {
+        nwarn("Healthcheck test command must be an array of strings");
+        cJSON_Delete(json);
+        return false;
     }
     
-    /* Parse Interval (now in seconds) */
+    if (strcmp(cmd_type->valuestring, "CMD") == 0 || strcmp(cmd_type->valuestring, "CMD-SHELL") == 0) {
+        /* Create test command array */
```

Comment:
Please validate the command string here, e.g. `strlen(cmd_value->valuestring) == 0 || strlen(cmd_value->valuestring) > 4096` ?